common/tools/dev-tool/src/util/checkWithTimeout.ts

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license
+
+import { createPrinter } from "./printer";
+const log = createPrinter("check-with-timeout");
+
+/**
+ *  - Maximum wait duration for the expected event to happen = `10000 ms`(default value is 10 seconds)(= maxWaitTimeInMilliseconds)
+ *  - Keep checking whether the predicate is true after every `1000 ms`(default value is 1 second) (= delayBetweenRetriesInMilliseconds)
+ */
+export async function checkWithTimeout(
```

Comment:
This method is in principle a `retry`. I think it could be good to name it `retryWithTimeout` so that, later on, if we decide to refactor the retry methods, we can refactor this as well 🙂

Reply:
It only checks and doesn't return anything.
This method has some history, I added the exact same method in service-bus in 2019.
Copied it here. 

The name was decided in 2019. 😛
Reference here - https://github.com/Azure/azure-sdk-for-js/pull/1214#discussion_r257094432